packages/contracts/contracts/libs/v0.8.x/minter-libs/SettlementExpLib.sol

```diff
@@ -0,0 +1,381 @@
+     * project's auction artifically sold out at a price higher than the
+     * project's base price, and the artist wishes to allow settlement at a
+     * lower price (and thus lower settlement revenues, increasing purchaser
+     * returned settlement funds by the minter to collectors).
+     * @param _projectId Project ID to reduce auction sellout price for.
+     * @param _newSelloutPrice New sellout price to set for the auction. Must
+     * be less than the current sellout price, gte base price.
+     */
+    function adminEmergencyReduceSelloutPrice(
+        uint256 _projectId,
+        address _coreContract,
+        uint128 _newSelloutPrice,
+        SettlementAuctionProjectConfig storage _settlementAuctionProjectConfig,
+        MaxInvocationsLib.MaxInvocationsProjectConfig
+            storage _maxInvocationsProjectConfig,
+        DAExpLib.DAProjectConfig storage _DAProjectConfig
+    ) internal {
+        // CHECKS
+        require(
+            !_settlementAuctionProjectConfig.auctionRevenuesCollected,
+            "Only before revenues collected"
+        );
+
+        // refresh max invocations, updating any local values that are
+        // illogical with respect to the current core contract state, and
+        // ensuring that local hasMaxBeenInvoked is accurate.
+        MaxInvocationsLib.refreshMaxInvocations(
+            _projectId,
+            _coreContract,
+            _maxInvocationsProjectConfig
+        );
+        require(
+            _newSelloutPrice >= _DAProjectConfig.basePrice,
+            "Only gte base price"
+        );
+        // require max invocations has been reached
+        require(
+            _maxInvocationsProjectConfig.maxHasBeenInvoked,
+            "Auction must be complete"
+        );
+        // @dev no need to check that auction max invocations has been reached,
+        // because if it was, the sellout price will be zero, and the following
+        // check will fail.
+        require(
+            _newSelloutPrice <
+                _settlementAuctionProjectConfig.latestPurchasePrice,
+            "May only reduce sellout price"
+        );
+        // ensure _newSelloutPrice is non-zero
+        // @dev this is a redundant check since minter doesn't allow base price
+        // of zero, but is included in case that logic changes in the future.
+        // @dev no coverage else branch of following line because redundant
+        require(_newSelloutPrice > 0, "Only sellout prices > 0");
+        // EFFECTS
+        _settlementAuctionProjectConfig.latestPurchasePrice = _newSelloutPrice;
+    }
+
+    /**
+     * @notice Gets the net revenues from the project's auction, in wei, and
+     * marks the auction as having had its revenues collected.
+     * IMPORTANT - this affects state and marks revenues collected as true, so
+     * revenue distribution should be done immediately after this function is
+     * called.
```

Comment:
This smells wrong, why not move `_settlementAuctionProjectConfig.auctionRevenuesCollected = true;` to the place where revenue is actually distributed..?

Reply:
We definitely want to mark as true before performing any interactions. I also noticed that this can also affect state of `_settlementAuctionProjectConfig.latestPurchasePrice`.
One option would be to import split funds lib to this library, include isEngine as an input param to the function, and perform the interactions here. I think that seems like probably the cleanest route 🤔

Reply:
Implemented the solution described above, after convo w/ @lyaunzbe about this. Seemed cleanest to perform the effects+interactions within the settlement lib itself, which warranted an updated, more idiomatic function name. updated in [52c0135](https://github.com/ArtBlocks/artblocks-contracts/pull/928/commits/52c0135dd1e7ef9a36a24695d8d45137d8fe8a1c)

Reply:
Cc @jakerockland on the above ^^